python/pyspark/sql/functions/builtin.py

```diff
@@ -21314,6 +21314,109 @@ def make_timestamp(
         )
 
 
+@_try_remote_functions
+def try_make_timestamp(
+    years: "ColumnOrName",
+    months: "ColumnOrName",
+    days: "ColumnOrName",
+    hours: "ColumnOrName",
+    mins: "ColumnOrName",
+    secs: "ColumnOrName",
+    timezone: Optional["ColumnOrName"] = None,
+) -> Column:
+    """
+    ANSI-compliant version of make_timestamp function.
+    Try to create timestamp from years, months, days, hours, mins, secs and timezone fields.
+    The result data type is consistent with the value of configuration `spark.sql.timestampType`.
+    The function returns NULL on invalid inputs.
+
+    .. versionadded:: 4.0.0
+
+    Parameters
+    ----------
+    years : :class:`~pyspark.sql.Column` or str
```

Comment:
```suggestion
    years : :class:`~pyspark.sql.Column` or column name
```


I have some users complained that it is unclear whether `str` is a literal string or column name, so I am changing the pydocs (in other PRs).

For the new functions in this PR, let's clarify that the `str` is a column name.

Reply:
fixed.